File: yolo_backend/src/utils/response.js
const moment = require('moment');

/**
 * 统一API响应格式
 */
class ApiResponse {
  static success(data = null, msg = 'Success') {
    return {
      code: '000000',
      statusCode: 200,
      msg,
      data,
      timestamp: moment().format('YYYY-MM-DD HH:mm:ss.SSS')
    };
  }

  static error(msg = 'Error', statusCode = 500) {
    const errorCodes = {
      400: 'A00400',  // Bad Request
      401: 'A00401',  // Unauthorized
      403: 'A00403',  // Forbidden
      404: 'A00404',  // Not Found
      500: 'A00500',  // Internal Server Error
      502: 'A00502',  // Bad Gateway
      503: 'A00503',  // Service Unavailable
      504: 'A00504'   // Gateway Timeout
    };

    return {
      code: errorCodes[statusCode] || 'A00500',
      statusCode,
      msg,
      data: null,
      timestamp: moment().format('YYYY-MM-DD HH:mm:ss.SSS')
    };
  }

  static badRequest(msg = 'Bad Request') {
    return this.error(msg, 400);
  }

  static unauthorized(msg = 'Unauthorized') {
    return this.error(msg, 401);
  }

  static forbidden(msg = 'Forbidden') {
    return this.error(msg, 403);
  }

  static notFound(msg = 'Not Found') {
    return this.error(msg, 404);
  }

  static serverError(msg = 'Internal Server Error') {
    return this.error(msg, 500);
  }
}

module.exports = ApiResponse;

File: sniper-yolo-backend/app/utils/response.py
"""统一响应格式工具 - 返回标准字典格式"""
from datetime import datetime
from typing import Any, Dict

def success_response(data: Any = None, msg: str = "Success", status_code: int = 200) -> Dict[str, Any]:
    """创建成功响应 - 支持自定义状态码"""
    return {
        "code": "000000",
        "statusCode": status_code,
        "msg": msg,
        "data": data,
        "timestamp": datetime.now().strftime("%Y-%m-%d %H:%M:%S")
    }

def error_response(code: str, status_code: int, msg: str) -> Dict[str, Any]:
    """创建错误响应"""
    return {
        "code": code,
        "statusCode": status_code,
        "msg": msg,
        "data": None,
        "timestamp": datetime.now().strftime("%Y-%m-%d %H:%M:%S")
    }